Description:

The 'TestStaticParams' test fails, as it creates a link to /bin/echo as /tmp/with space and attempts
to use this. This fails with a busybox based system, as busybox does not know which applet to run
and exits with an error.

This patch creates a small shell script as /tmp/with space, which simply runs 'echo'.

Upstream: yes

URL: https://github.com/adnanh/webhook/pull/518

diff -Nurp ../webhook-2.8.0-orig/webhook_test.go ./webhook_test.go
--- ../webhook-2.8.0-orig/webhook_test.go	2020-12-06 07:42:20.000000000 +0000
+++ ./webhook_test.go	2021-03-13 14:26:43.911681652 +0000
@@ -33,7 +33,8 @@ func TestStaticParams(t *testing.T) {
 	spHeaders["Accept"] = "*/*"

 	// case 2: binary with spaces in its name
-	err := os.Symlink("/bin/echo", "/tmp/with space")
+	d1 := []byte("#!/bin/sh\n/bin/echo\n")
+	err := ioutil.WriteFile("/tmp/with space", d1, 0755)
 	if err != nil {
 		t.Fatalf("%v", err)
 	}
